controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/systemflags/SystemFlagsDeployerTest.java

```diff
@@ -74,7 +81,81 @@ void deploys_flag_data_to_targets() throws IOException {
     }
 
     @Test
-    void dryrun_should_not_change_flags() throws IOException {
+    void deploys_partial_flag_data_to_targets() {
+        // default.json contains one rule with 2 conditions, one of which has a condition on the aws cloud.
+        // This condition IS resolved for a config server target, but NOT for a controller target, because FLAG_ID
+        // has the CLOUD dimension set.
+        deployFlags(Optional.empty(), "partial/default.json", Optional.of("partial/put-controller.json"), true, PutType.CREATE, FetchVector.Dimension.CLOUD);
+        deployFlags(Optional.empty(), "partial/default.json", Optional.empty(), false, PutType.NONE, FetchVector.Dimension.CLOUD);
+        deployFlags(Optional.of("partial/initial.json"), "partial/default.json", Optional.of("partial/put-controller-2.json"), true, PutType.UPDATE, FetchVector.Dimension.CLOUD);
```

Comment:
`put-controller-2.json` is identical to `put-controller.json`?

Reply:
Right, fixed.